osu.Game/Database/RealmWriteUsage.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using Realms;
+
+namespace osu.Game.Database
+{
+    public class RealmWriteUsage : IDisposable
+    {
+        public readonly Realm Context;
+        private readonly Action<RealmWriteUsage> usageCompleted;
+
+        public bool RollbackRequired { get; private set; }
+
+        public RealmWriteUsage(Realm context, Action<RealmWriteUsage> onCompleted)
+        {
+            Context = context;
+            usageCompleted = onCompleted;
+        }
+
+        /// <summary>
+        /// Whether this write usage will commit a transaction on completion.
+        /// If false, there is a parent usage responsible for transaction commit.
+        /// </summary>
+        public bool IsTransactionLeader;
+
+        private bool isDisposed;
+
+        protected void Dispose(bool disposing)
+        {
+            if (isDisposed) return;
+
+            isDisposed = true;
+
+            usageCompleted?.Invoke(this);
+        }
+
+        public void Rollback(Exception error = null)
+        {
+            RollbackRequired = true;
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        ~RealmWriteUsage()
+        {
+            Dispose(false);
+        }
```

Comment:
is there a need for this finaliser? are there unmanaged resources to be freed by this?

Reply:
I think it was just for safety (this has come across from `DatabaseWriteUsage`). Can be removed if we are 100% sure.

Reply:
It *looks* safe to remove as far as I can tell. The finaliser doesn't seem to be able to do anything anyway (`Dispose(bool)` early returns if it's called from finaliser).